### PR TITLE
Ticket4414 read access configs

### PR DESCRIPTION
**Pull request by adamb56789**
### Description of work

Changed editing of configurations and components to also allow viewing when in read-only. This caused a bug where trying to view a component or configuration sometimes caused an error message (no reliable way to reproduce this). Tracked this down to Awaited, which was waiting for the full maximum length of time when it shouldn't have been, fixed this and also increased the maximum wait time from 1 to 2 seconds.
We found some files that were not being used, so deleted them.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/4414

### Acceptance criteria

- [x] With edit access, the functionality of editing the configurations or components is unchanged
- [x] It is possible to view all of the configurations and components without edit access
- [x] An error is never displayed when trying to view a configuration or component
- [x] Unit tests are sensible and pass

### Unit tests

Created some unit tests for the functionality of awaited.

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Do the changes function as described and is it robust?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [x] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.model.tests/src/uk/ac/stfc/isis/ibex/model/tests/AwaitedTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import uk.ac.stfc.isis.ibex.model.Awaited;
+import uk.ac.stfc.isis.ibex.model.UpdatedValue;
+
+
+@SuppressWarnings("checkstyle:methodname")
```
Comment (davidkeymer) [inline base/uk.ac.stfc.isis.ibex.model.tests/src/uk/ac/stfc/isis/ibex/model/tests/AwaitedTest.java]: Missing docstring.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/helpers/ConfigHelper.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 	
     protected ConfigurationViewModels configurationViewModels;
     protected Shell shell;
+    protected static final int MAX_SECONDS_TO_WAIT = 2;
```
[the remark this answers is not in the document]
Comment (adamb56789) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/helpers/ConfigHelper.java]: I've set it to 10.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditConfigHandler.java [hunk 3]
```diff
@@ -1,71 +1,160 @@
...
 
 /**
- * Handler for the edit configuration menu command.
+ * The handler class for editing configurations.
+ * 
+ * It sets the menu labels, and opens the dialogue for editing or viewing the
+ * configurations.
  */
-public class EditConfigHandler extends DisablingConfigHandler<Configuration> {
+public class EditConfigHandler extends ConfigHandler<Configuration> {
```
[the remark this answers is not in the document]
Comment (adamb56789) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditConfigHandler.java]: I've made a new base class which both of these now inherit from.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditComponentHandler.java [hunk 4]
```diff
@@ -1,59 +1,148 @@
...
+            helper = new EditComponentHelper(shell, SERVER);
+            titleText = EDIT_TITLE;
+        } else {
+            helper = new ViewComponentHelper(shell);
+            titleText = VIEW_TITLE;
+        }
+
         ConfigSelectionDialog selectionDialog =
-                new ConfigSelectionDialog(shell, TITLE, SERVER.componentsInfo().getValue(), true, true);
+                new ConfigSelectionDialog(shell, titleText, SERVER.componentsInfo().getValue(), false, true);
```
[the remark this answers is not in the document]
Comment (adamb56789) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditComponentHandler.java]: Not sure why it was ```false```, I think it changed some menu text. Changed it back to ```true```.

**Comment by adamb56789** [comment]
I've made all of the requested changes.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditComponentHandler.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (davidkeymer) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditComponentHandler.java]: Functionality seems OK.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditConfigHandler.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidkeymer) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/EditConfigHandler.java]: Class checked and coding good.

File: base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/helpers/ConfigHelper.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidkeymer) [reply base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/helpers/ConfigHelper.java]: Good idea.  We'll check value is appropriate during system testing.

**Pull request merged**